Cython/Compiler/ModuleNode.py

```diff
@@ -972,7 +972,8 @@ def generate_gcc33_hack(self, env, code):
 
     def generate_typedef(self, entry, code):
         base_type = entry.type.typedef_base_type
-        if base_type.is_numeric:
+        enclosing_scope = entry.scope
+        if base_type.is_numeric and not enclosing_scope.is_cpp_class_scope:
```

Comment:
Here we are in the C++ code generation part.

The base behaviour is to put all the numeric type definitions early in the generated C++ code (the numeric_typedefs section), regardless of where they are defined in the cython code.

When we are generating nested numeric type definitions, i.e. we are defining a numeric type inside a cppclass, we don't want that numeric type to be defined elsewhere than inside the cppclass namespace. So, when we are in a cppclass, we avoid escaping numeric types to the numeric_typedefs section.